### PR TITLE
DAOS-11276 engine: use ULT for IV retry

### PR DESCRIPTION
**Pull request by wangdi1**
Use ULT to retry IV operation to avoid stack
overflow during endless retry.

Signed-off-by: Di Wang <di.wang@intel.com>

### CODE REVIEW

**Comment by wangdi1** [comment]
Do not need this patch.

File: src/engine/server_iv.c [hunk 1]
```diff
@@ -1105,16 +1117,35 @@ iv_op_async(struct ds_iv_ns *ns, struct ds_iv_key *key, d_sg_list_t *value,
...
 	ds_iv_ns_get(ns);
 	ult_arg->ns = ns;
 	ult_arg->opc = opc;
+
+	/* Synchronous mode, let's create the eventual to wait */
+	if (sync == NULL || sync->ivs_mode != CRT_IV_SYNC_LAZY) {
+		rc = ABT_eventual_create(sizeof(*status), &done_eventual);
+		if (rc)
+			return rc;
+		ult_arg->eventual = done_eventual;
```
Comment (liuxuezhao) [inline src/engine/server_iv.c:1132]: if "sync->ivs_mode == CRT_IV_SYNC_LAZY", seems no place to create ult_arg->eventual? Then in "iv_op_ult() -> ABT_eventual_set(ult_arg->eventual, NULL, rc)" the ult_arg->eventual is NULL?

File: src/engine/server_iv.c [hunk 2]
```diff
@@ -1066,26 +1066,38 @@ _iv_op(struct ds_iv_ns *ns, struct ds_iv_key *key, d_sg_list_t *value,
...
 {
-	struct iv_op_ult_arg *ult_arg = arg;
+	struct iv_op_ult_arg	*ult_arg = arg;
+	int			rc;
 
-	D_ASSERT(ult_arg->iv_sync.ivs_mode == CRT_IV_SYNC_LAZY);
 	/* Since it will put LAZY sync in a separate and asynchronous ULT, so
 	 * let's use EAGER mode in CRT to make it simipler.
 	 */
 	ult_arg->iv_sync.ivs_mode = CRT_IV_SYNC_EAGER;
```
Comment (liuxuezhao) [inline src/engine/server_iv.c:1075]: if originally sync->ivs_mode == CRT_IV_SYNC_NONE, that IV OP got retry (for example DER_GERVER), then the retry will call iv_op_by_ult() -> iv_op_ult() here will force it to work in SYNC_EAGER mode, look that is unnecessary for the sync?

**Comment by liuxuezhao** [comment]
oh, just saw it is closed, replace by other patch?

**Comment by wangdi1** [comment]
> oh, just saw it is closed, replace by other patch?

oh, we do not need this patch anymore. Just thought this is not an issue at all.

**Pull request closed without merging**